src/M6Web/Bundle/ElasticsearchBundle/Connection/TookExtractor.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace M6Web\Bundle\ElasticsearchBundle\Connection;
+
+/**
+ * TookExtractor
+ */
+trait TookExtractor
+{
+    /**
+     * Extract took from response
+     *
+     * @param array $response
+     *
+     * @return int|null
+     */
+    protected function extractTookFromResponse(array $response)
+    {
+        if (!isset($response['text'])) {
+
+            return null;
+        }
+
+        $matches = array();
+        if (preg_match('/\"took\":(\d+)/', $response['text'], $matches)) {
+
+            return (int) $matches[1];
+        }
+
+        return null;
+    }
```

Comment:
The construction of this method is strange : 
```
if(some test) {
  // non standard use-case
}

if(other test) {
  // standard use-case
}

// fallback
```

To be consistent, you should use a major direction, and exit.
2 way to write the same code : 

```php
protected function extractTookFromResponse(array $response)
{
    if (!isset($response['text'])) {
        return null;
    }

    $matches = array();
    if (!preg_match('/\"took\":(\d+)/', $response['text'], $matches)) {
        return null;
    }

    return return (int) $matches[1];;
}
```

```php
protected function extractTookFromResponse(array $response)
{
    if (isset($response['text'])) {
        $matches = array();
        if (preg_match('/\"took\":(\d+)/', $response['text'], $matches)) {
            return return (int)$matches[1];
        }
    }

    return null;
}
```